x-pack/plugins/aiops/public/components/change_point_detection/change_point_detection_context.tsx

```diff
@@ -0,0 +1,275 @@
+        query: savedSearchQuery?.searchString ?? '',
+        language: savedSearchQuery?.queryLanguage ?? 'kuery',
+      }
+    );
+  }, [savedSearchQuery, requestParamsFromUrl.query]);
+
+  const requestParams = useMemo(() => {
+    const params = { ...requestParamsFromUrl };
+    if (!params.fn) {
+      params.fn = DEFAULT_AGG_FUNCTION;
+    }
+    if (!params.metricField && metricFieldOptions.length > 0) {
+      params.metricField = metricFieldOptions[0].name;
+    }
+    if (!params.splitField && splitFieldsOptions.length > 0) {
+      params.splitField = splitFieldsOptions[0].name;
+    }
+    params.interval = bucketInterval?.expression!;
+    return params;
+  }, [requestParamsFromUrl, metricFieldOptions, splitFieldsOptions, bucketInterval]);
+
+  const updateFilters = useCallback(
+    (update: Filter[]) => {
+      filterManager.setFilters(update);
+    },
+    [filterManager]
+  );
+
+  useMount(() => {
+    setResultFilter(filterManager.getFilters());
+    const sub = filterManager.getUpdates$().subscribe(() => {
+      setResultFilter(filterManager.getFilters());
+    });
+    return () => {
+      sub.unsubscribe();
+    };
+  });
+
+  useEffect(
+    function syncFilters() {
+      const globalFilters = filterManager?.getGlobalFilters();
+      if (requestParamsFromUrl.filters) {
+        filterManager.setFilters(requestParamsFromUrl.filters);
+      }
+      if (globalFilters) {
+        filterManager?.addFilters(globalFilters);
+      }
+    },
+    [requestParamsFromUrl.filters, filterManager]
+  );
+
+  const combinedQuery = useMemo(() => {
+    const mergedQuery = createMergedEsQuery(resultQuery, resultFilters, dataView, uiSettings);
+    if (!Array.isArray(mergedQuery.bool?.filter)) {
+      if (!mergedQuery.bool) {
+        mergedQuery.bool = {};
+      }
+      mergedQuery.bool.filter = [];
+    }
+
+    mergedQuery.bool!.filter.push({
+      range: {
+        [dataView.timeFieldName!]: {
+          from: timeRange.from,
+          to: timeRange.to,
+        },
+      },
+    });
+
+    return mergedQuery;
+  }, [resultFilters, resultQuery, uiSettings, dataView, timeRange]);
+
+  const {
+    results: annotations,
+    isLoading: annotationsLoading,
+    progress,
+    pagination,
+  } = useChangePointResults(requestParams, combinedQuery);
+
+  if (!bucketInterval) return null;
+
+  const value = {
+    isLoading: annotationsLoading,
+    progress,
+    timeBuckets,
+    requestParams,
+    updateRequestParams,
+    metricFieldOptions,
+    splitFieldsOptions,
+    annotations,
+    bucketInterval,
+    resultFilters,
+    updateFilters,
+    resultQuery,
+    pagination,
+  };
+
+  return (
+    <ChangePointDetectionContext.Provider value={value}>
+      {children}
+    </ChangePointDetectionContext.Provider>
+  );
+};
+
+export function useChangePontDetectionContext() {
+  return useContext(ChangePointDetectionContext);
+}
+
+export function useRequestParams() {
+  return useChangePontDetectionContext().requestParams;
```

Comment:
Typo, should be `useChangePointDetectionContext`.

Reply:
Fixed in https://github.com/elastic/kibana/pull/144093/commits/92ce0632a18f52338f038b66937d45e2fa7facf6